### PR TITLE
Implement East_Asian_Width and Line_Break Unicode properties

### LINKED ISSUE #1212: Implement Unicode properties required by line breaker (UAX 14)
We need `Line_Break` at runtime to replace `UAX14_PROPERTY_TABLE`, and also `East_Asian_Width` when generating line break rules.

https://www.unicode.org/reports/tr14/#Properties

### PR DESCRIPTION
**Pull request by aethanyc**
Fixed #1212.

### CODE REVIEW

**Review by markusicu — APPROVED** [review]
lgtm

**Comment by aethanyc** [comment]
@markusicu Thank you for the review.

`provider/testdata/data/testdata.postcard` is modified on 'main' branch, so I have to merge 'main' and rerun `cargo make testdata` again to update `testdata.postcard`.

File: components/properties/src/maps.rs [hunk 1]
```diff
@@ -96,6 +96,54 @@ where
...
+///
+/// let provider = icu_testdata::get_provider();
+/// let payload = maps::get_east_asian_width(&provider).expect("The data should be valid!");
+/// let eaw = &payload.get().code_point_trie;
+///
+/// assert_eq!(eaw.get('ｱ' as u32), EastAsianWidth::Halfwidth); // U+FF71: Halfwidth Katakana Letter A
+/// assert_eq!(eaw.get('ア' as u32), EastAsianWidth::Wide); //U+30A2: Katakana Letter A
+/// ```
+///
+/// [`CodePointTrie`]: icu_codepointtrie::codepointtrie::CodePointTrie
```
Comment (sffc) [inline components/properties/src/maps.rs:115]: Nit, here and elsewhere: Please update these comments to point to `icu_codepointtrie::CodePointTrie`; I hoisted it to the top level

**Review by sffc — APPROVED** [review]
LGTM with a small cleanup comment

File: components/properties/src/maps.rs (hunk 1 above)
Comment (aethanyc) [reply components/properties/src/maps.rs]: Added a commit to address this.

File: components/properties/src/props.rs [hunk 2]
```diff
@@ -426,6 +430,87 @@ impl Script {
...
+    pub const JL: LineBreak = LineBreak(33); // name="JL"
+    pub const JT: LineBreak = LineBreak(34); // name="JT"
+    pub const JV: LineBreak = LineBreak(35); // name="JV"
+    pub const CloseParenthesis: LineBreak = LineBreak(36); // name="CP"
+    pub const ConditionalJapaneseStarter: LineBreak = LineBreak(37); // name="CJ"
+    pub const HebrewLetter: LineBreak = LineBreak(38); // name="HL"
+    pub const RegionalIndicator: LineBreak = LineBreak(39); // name="RI"
+    pub const EBase: LineBreak = LineBreak(40); // name="EB"
+    pub const EModifier: LineBreak = LineBreak(41); // name="EM"
+    pub const ZWJ: LineBreak = LineBreak(42); // name="ZWJ
```
Comment (echeran) [inline components/properties/src/props.rs:511]:
```suggestion
    pub const ZWJ: LineBreak = LineBreak(42); // name="ZWJ"
```

**Review by echeran — DISMISSED** [review]
LGTM.  I left a suggested change for a tiny nitpick.

**Review by echeran — APPROVED** [review]

**Pull request merged**